CMakeLists.txt

```diff
@@ -1,10 +1,275 @@
+endfunction()
+
+# Windows detection function
+function(configure_windows_target target)
+    if(NOT WIN32 OR NOT MSVC)
+        return()
+    endif()
+
+    string(REGEX REPLACE "/W[0-4]" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+
+# These are the survivors that earned their right to silence
+    target_compile_options(${target} PRIVATE
+        /MP${CPU_CORES}
+        /bigobj
+        /GR 
+        /EHsc
+        /nologo
+        /WX-
+        /Zc:preprocessor
+        /Zc:__cplusplus
+        /Zc:externConstexpr
+        /Zc:throwingNew
+        /Zc:rvalueCast
+        /FI"cstddef"
+        /FI"cstdint"
+        /FI"atomic"
+        $<$<CONFIG:Debug>:/MDd /Od /Ob1 /RTC1>
+        $<$<NOT:$<CONFIG:Debug>>:/MD /O2 /Ob2 /DNDEBUG>
+        # Critical warning suppressions
+        /wd4005  # Macro redefinition
+        /wd4273  # Inconsistent dll linkage
+        /wd4141  # 'inline' used more than once
+        # Other warning disables
+        /wd4068  # Unknown pragma
+        /wd4100  # Unreferenced parameter
+        /wd4127  # Conditional expression
+        /wd4189  # Local variable not referenced
+        /wd4201  # Nameless struct/union
+        /wd4244  # Conversion possible loss of data
+        /wd4245  # Signed/unsigned mismatch
+        /wd4267  # Conversion from size_t
+        /wd4305  # Truncation
+        /wd4310  # Cast truncates constant
+        /wd4311  # Pointer truncation
+        /wd4324  # Structure padding
+        /wd4458  # Declaration hides member
+        /wd4464  # Relative include path
+        /wd4505  # Unreferenced function removed
+        /wd4514  # Unreferenced inline removed
+        /wd4530  # Exception handler
+        /wd4611  # setjmp interaction
+        /wd4702  # Unreachable code
+        /wd4710  # Function not inlined
+        /wd4711  # Selected for inline
+        /wd4820  # Padding added
+        /wd4996  # Deprecated functions
+        /wd5026  # Move ctor deleted
+        /wd5027  # Move assign deleted
+        /wd5045  # Spectre mitigation
+    )
+
+    target_compile_definitions(${target} PRIVATE
+        WIN32_LEAN_AND_MEAN
+        NOMINMAX
+        _CRT_SECURE_NO_WARNINGS
+        _CRT_NONSTDC_NO_WARNINGS
+        _SCL_SECURE_NO_WARNINGS
+        _HAS_EXCEPTIONS=1
+    )
+
+    set_target_properties(${target} PROPERTIES
+        MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL"
+        VS_GLOBAL_CLToolChain "v143"
+        VS_GLOBAL_PlatformToolset "v143"
+        VS_GLOBAL_DisableSpecificWarnings "MSB8027;MSB8065;4244;4267;4996;4273;4141;4005;4141" 
+        VS_GLOBAL_TreatWarningAsError "false"
+        VS_GLOBAL_WarningLevel "0"
+    )
+
+    configure_cpu_optimized_unity_build(${target})
+endfunction()
+
+# Linux detection function
+function(configure_linux_target target)
+    if(WIN32)
+        return()
+    endif()
+
+    target_compile_options(${target} PRIVATE
+        -Wall
+        -Wextra
+        -Wpedantic
+        -Wno-unknown-pragmas
+        -fPIC
```

Comment:
This shouldn't be needed because we use the POSITION_INDEPENDENT_CODE from Cmake which ensures it is correctly set for all compilers cmake supports.